src/mbgl/renderer/painter.hpp

```diff
@@ -178,7 +177,7 @@ class Painter : private util::noncopyable {
     }();
 
 private:
-    TransformState state;
+    Transform& transform;
```

Comment:
If it helps, I think it's fine to have `Painter::render` take a `Transform` by copy rather than a `TransformState`, so that this member becomes `Transform transform`. In that case I would still keep a `TransformState& state` member around that references `transform.currentState()` to avoid needing to change all the paint implementation methods.

Long term I think `Transform` and `TransformState` will merge.